website/docs/concepts/event-bridge/index.md

```diff
@@ -28,6 +26,262 @@ To jump right into building, see the [`EventBus`](./event-bus.md) documentation.
 
 An instance of an [`EventBus`](./event-bus.md) ingest events and routes them to downstream integrations according to Rules created by the user. Events are sent to downstream services such as Lambda Functions, Step Functions, or a third party (non-AWS) API. Sources of events include other AWS Resources or third party (non-AWS) SaaS products, e.g. a Slack webhook.
 
-## Default Event Bus
+## Declare an Event Type
+
+Functionless supports well typed events, lets add our event schema to Typescript.
+
+```ts
+interface UserDetails {
+  id?: string;
+  name: string;
+  age: number;
+  interests: string[];
+}
+
+interface UserEvent
+  extends functionless.EventBusEvent<
+    UserDetails,
+    // We can provide custom detail-types to match on
+    "Create" | "Update" | "Delete"
+  > {}
+```
+
+## Create an Rule
+
+Now that you have a wrapped `EventBus`, lets add some rules.
+
+Functionless lets you write logic in Typescript on the type safe event.
+
+Lets match all of the `Create` or `Update` events with one rule and another rule for `Delete`s.
+
+```ts
+const createOrUpdateEvents = bus.when(
+  this,
+  "createOrUpdateRule",
+  (event) =>
+    event["detail-type"] === "Create" || event["detail-type"] === "Update"
+);
+const deleteEvents = bus.when(
+  this,
+  "deleteRule",
+  (event) => event["detail-type"] === "Delete"
+);
+```
+
+We also want to do something special when we get a new cat lover who is between 18 and 30 years old, lets make another rule for those.
+
+```ts
+const catPeopleEvents = bus.when(
+  (event) =>
+    event["detail-type"] === "Create" &&
+    event.detail.interests.includes("CATS") &&
+    event.detail.age >= 18 &&
+    event.detail.age < 30
+);
+```
+
+Rules can be further refined by calling `when` on a Functionless `Rule`.
+
+```ts
+// Cat people who are between 18 and 30 and do not also like dogs.
+catPeopleEvents.when((event) => !event.detail.interests.includes("DOGS"));
+```
+
+## Transform the event before sending to some services like `Lambda` Functions.
+
+We have two lambda functions to invoke, one for create or updates and another for deletes, lets make those.
```

Comment:
you could probably simplify this section by linking to the function docs. 80% of this is redundant, the meat is the `map` calls at the bottom

Reply:
Updated to use the new Functionless function constructors.